src/Runner.Common/JobServer.cs

```diff
@@ -335,6 +336,14 @@ public Task CreateResultsStepLogAsync(string planId, string jobId, Guid stepId,
             throw new InvalidOperationException("Results client is not initialized.");
         }
 
+        public Task CreateResultsJobLogAsync(string planId, string jobId, string file, bool finalize, bool firstBlock, long lineCount, CancellationToken cancellationToken)
+        {
+            if (_resultsClient != null)
+            {
+                return _resultsClient.UploadResultsJobLogAsync(planId, jobId, file, finalize, firstBlock, lineCount, cancellationToken: cancellationToken);
+            }
+            throw new InvalidOperationException("Results client is not initialized.");
```

Comment:
Double checking this time to make sure the exception will be caught 😄

Reply:
Since we're dual-uploading, do we even need this exception? I wonder if we could just add some tracing here instead.

Reply:
We actually caught the exception before as well - the issue was in the handling part. 😓 

https://github.com/actions/runner/blob/main/src/Runner.Common/JobServerQueue.cs#L520

Here we wanted to log a telemetry for all the failures, but on GHES server doesn't understand this record and errored out.

Reply:
Now we are properly guarded and if the feature flag is not on, nothing will be put on the Results processing queue.

I also tested these changes against a GHES instance.

Reply:
@brittanyellich we shouldn't get to this point if we aren't going to upload to results, so I feel the exception is still appropriate.